File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/model/CardsRemoteDao.java
package me.nathanp.magiccreatures.model;

import android.util.Log;

import com.google.firebase.firestore.FirebaseFirestore;

import java.util.List;

import androidx.lifecycle.LiveData;

class CardsRemoteDao {
    private static final String TAG = "CardsRemoteDao";

    private LiveFirestoreCollection<Card> liveCardCollection;

    public CardsRemoteDao() {
        Log.d(TAG, Paths.getCardCollectionPath());
        liveCardCollection = new LiveFirestoreCollection<>(FirebaseFirestore.getInstance().collection(Paths.getCardCollectionPath()), Card.class);
    }

    public LiveData<List<Card>> getAllCards() {
        return liveCardCollection;
    }
}

File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/model/CardsRemoteDao.kt
package me.nathanp.magiccreatures.model

import androidx.lifecycle.LiveData
import com.google.firebase.firestore.FirebaseFirestore

internal class CardsRemoteDao {
    private val TAG = javaClass.simpleName
    private val liveCardCollection = LiveFirestoreCollection(FirebaseFirestore.getInstance().collection(Paths.cardCollectionPath), Card::class.java)
    val allCards: LiveData<List<Card>>
        get() = liveCardCollection
}